### ISSUE TITLE
Attention/memory example

### ISSUE THREAD

**Issue by dmccloskey** [dmccloskey/EvoNet#52]
### References
- https://distill.pub/2016/augmented-rnns/
- https://medium.com/syncedreview/memory-attention-sequences-8522f531dd43
- https://jalammar.github.io/illustrated-transformer/
- https://github.com/tensorflow/tensor2tensor/blob/23bd23b9830059fbc349381b70d9429b5c40a139/tensor2tensor/layers/common_attention.py
- http://nlp.seas.harvard.edu/2018/04/03/attention.html#background

**Comment by dmccloskey** [comment]
YonghuiWu, Mike Schuster, Zhifeng Chen, Quoc V. Le, Mohammad Norouzi,Wolfgang Macherey,
Maxim Krikun, Yuan Cao, Qin Gao, Klaus Macherey, Jeff Klingner, Apurva Shah, Melvin Johnson,
Xiaobing Liu, Łukasz Kaiser, Stephan Gouws, Yoshikiyo Kato, Taku Kudo, Hideto Kazawa,
Keith Stevens, George Kurian, Nishant Patil, Wei Wang, Cliff Young, Jason Smith, Jason Riesa,
Alex Rudnick, Oriol Vinyals, Greg Corrado, Macduff Hughes, and Jeffrey Dean. Google’s neural
machine translation system: Bridging the gap between human and machine translation. arXiv
preprint arXiv:1609.08144, 2016.

ModelBuilder::addAdditiveAttention (GNMT)

Noam Shazeer, Azalia Mirhoseini, Krzysztof Maziarz, Andy Davis, Quoc Le, Geoffrey Hinton,
and Jeff Dean. Outrageously large neural networks: The sparsely-gated mixture-of-experts layer.
arXiv preprint arXiv:1701.06538, 2017.

ModelBuilder::addGNMTAttentionMod

ModelBuilder::addAdditiveAttention 
- score between source and target where score can be the following:
  - dot: Transpose(ht-1)*et
  - general: Transpose(ht-1) * W * et
  - concat or additive: Transpose(v)*tanh(W[ht-1;et])
- alpha = softmax score
- c = context vector Sum[t=0,n](alphat * et)
- a = attention = tanH(W(c;target) [Not sure this is needed]
